lib/backend_version.rb

```diff
@@ -1,27 +1,82 @@
 # frozen_string_literal: true
 
-# TODO: Unify this with Logic::Backend::Version in lib/logic/backend.rb
-
 class BackendVersion < ActiveSupport::StringInquirer
-  V1 = '1'
-  V2 = '2'
-  OAUTH = 'oauth'
 
-  VERSIONS = [V1, V2, OAUTH]
+  VERSIONS = { 'v1' => '1', 'v2' => '2', 'oauth' => 'oauth', 'oidc' => 'oidc' }.freeze
+  private_constant :VERSIONS
 
-  def initialize(value)
-    super(value.to_s) if value
+  class << self
+
+    VERSIONS.each_key do |method|
+      define_method "#{method}_usable?" do |_|
+        true
+      end
+      define_method "#{method}_visible?" do |_|
+        true
+      end
+    end
+
+    def oidc_usable?(_)
+      false
+    end
+
+    def oidc_visible?(service)
+      service.account.provider_can_use?(:apicast_oidc) &&
+        (service.proxy || service.build_proxy).apicast_configuration_driven
+    end
+
+    def oauth_visible?(service)
+      service.backend_version.oauth? && !service.oidc?
+    end
+
+    def usable_versions(service:)
+      VERSIONS.map do |method, name|
```

Comment:
`select` is better in this case

Reply:
then i would have to do:

```
VERSIONS.select do |method, name|
  name if public_send("#{method}_usable?", service)
end.values
```

tomato potato i would say.